File: packages/cdk/lambda/preTokenGeneration.ts
import {
  PreTokenGenerationTriggerEvent,
  PreTokenGenerationTriggerHandler,
} from 'aws-lambda';

export const handler: PreTokenGenerationTriggerHandler = async (
  event: PreTokenGenerationTriggerEvent
) => {
  const tenantId = event.request.userAttributes['custom:tenant_id'] || '';

  event.response = {
    claimsOverrideDetails: {
      claimsToAddOrOverride: {
        tenant_id1: tenantId,
        'https://aws.amazon.com/tags': JSON.stringify({
          principal_tags: {
            tenant_id: [tenantId],
          },
        }),
      },
    },
  };

  return event;
};

File: packages/cdk/lambda/pre_token_generation.py
def lambda_handler(event, context):
    event["response"]["claimsAndScopeOverrideDetails"] = {
        "idTokenGeneration": {
            "claimsToAddOrOverride": {
                "https://aws.amazon.com/tags": {
                    "principal_tags": {"TenantID": ["<TENANT_ID>"]}
                }
            }
        }
    }
    return event
